ml4gw/transforms/injection.py

```diff
@@ -382,14 +383,17 @@ def sample(
         )
 
         if self.snr is not None:
-            target_snrs = self._sample_source_param(self.snr, idx, N)
-            rescaled_responses = gw.reweight_snrs(
-                ifo_responses,
-                target_snrs,
-                backgrounds=self.background,
-                sample_rate=self.sample_rate,
-                highpass=self.mask,
+            snrs = gw.compute_network_snr(
+                ifo_responses, self.background, self.sample_rate, self.mask
             )
+            if (self.snr == -1).all():
```

Comment:
So is the idea here to pass a tensor of all `-1`s at `__init__` time? It might be more intuitive to just be able to pass a literal `-1`. I know that torch is finnicky about comparing tensors to other stuff, so it might be easiest to include a type check here: `if isinstance(self.snr, int) and self.snr == -1:` (though I guess since `-1` is the only `int` we allow this is probably redundant)

Reply:
I think we'll want to do a special case of `if name == "snr" and isinstance(param, int):` during `__init__`. Doing explicit `int` checking isn't necessarily the most recommended pythonic behavior, but it's easier than adding in a separate catch for `TypeError` during the `len` check at line 179

Reply:
Yeah, this was to account for the option that `self.snr` was an array of SNRs. The type checking makes more sense though

Reply:
Am I right in saying that we should have `SourceParameter = Union[np.ndarray, torch.Tensor, Distribution]`? Just because of how we use it in `_sample_source_param`.